lessons/3.mdx

```diff
@@ -0,0 +1,1044 @@
+
+</aside>
+
+We delete some files so we start fresh:
+
+```bash
+rm contracts/Lock.sol
+rm scripts/deploy.js
+rm test/Lock.js
+```
+
+In a Hardhat project, the default folders are supposed to be for:
+
+- `contracts/` is where the source files for your contracts should be.
+- `scripts/` is where simple automation scripts go.
+- `test/` is where your tests should go.
+
+### Let’s start coding!
+
+Fire up your code editor and let’s start hacking. I’m using VSCode, so I run
+`code .` in my terminal.
+
+Let’s create an empty file named `TierNFT.sol` inside the `contracts/` folder
+and copy this code inside:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+contract TierNFT {
+}
+```
+
+Now that we have our License, the version of solidity we will be using and the
+contract set up, we can add the logic and the state.
+
+## What do we need?
+
+As we are creating tiers (or categories/levels) for our NFTs, we need to store
+information about tiers on our contracts.
+
+Let’s write our smart contract within four steps. We will try to go step by
+step.
+
+### Inheriting OpenZeppelin ERC721 and adding a Mint Function
+
+Let’s get started with a simple contract. We will inherit OpenZeppelin ERC721
+contract again.
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+contract TierNFT is ERC721 {
+
+    constructor(string memory _name, string memory _symbol)
+        ERC721(_name, _symbol) {}
+
+}
+```
+
+We have a constructor on our contract. We need to have this constructor as we
+are inheriting ERC721 from OpenZeppelin implementation. If we didn’t have the
+constructor then the compiler would have thrown us errors.
+
+Now let’s go ahead and add a mint function to our contract. We will add a
+function that only uses `_safeMint`
+
+<br/>
+<details>
+<summary>**Our code will look something like this**</summary>
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol'
+
+contract TierNFT is ERC721 {
+
+    uint256 public totalSupply;
+
+    constructor(string memory _name, string memory _symbol)
+        ERC721(_name, _symbol)
+    {}
+
+    function mint() public payable {
+        totalSupply++;
+        _safeMint(msg.sender, totalSupply);
+    }
+
+}
+
+````
+</details>
+<br/>
+
+
+### Adding Tiers and Using Them in the Mint Function
+
+Now it’s time for us to add our tiers. Just as we explained before each tiers
+will have different values as the service subscription. Now let’s go ahead and
+add the tiers and assign their values.
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+string constant TIER_NAME_0 = "NOOB";
```

Comment:
Hey @7i7o I noticed at the beginning of lesson, we show example of Netflix for tiers, but then we use NOOB, DEV and HACKER in the contract as the tiers.

Do we need to tie these together better? (Either change overview to mention NOOB, DEV, HACKER; or change contract to be Netflix)

Reply:
I think we should match the lesson explanation to make it more real-world related.
I think we should change the contract definitions.

Reply:
Perfect thank you @7i7o!